### ISSUE TITLE
High memory usage on OAI-PMH interface during and after harvesting

### ISSUE THREAD

**Issue by ChrisBethgster** [DSpace/DSpace#8846]
Version: 6.4

**To Reproduce**
1. Harvest with a client the complete dspace repository with loop over all identifiers with the method GetRecord on OAI.

After a big size of harvested items (3000 items) the memory usage is increased on server. But the CPU usage is yet low.
After a big, big size of harvested items (5000 items) the memory usage is increased and the CPU usage is **continuous HIGH**.
The GC running continously and blocks the application (timeouts on OAI-PMH, UI,...). The memory usage retained increased.

This is a new behaviour and doesn't exists in version 6.3.

**Comment by ChrisBethgster** [comment]
Difference in the OAI implementation between 6.3 and 6.4 is

#8269

**Comment by ChrisBethgster** [comment]
Memory dump version 6.4 with Saxon transformer...

![image](https://github.com/DSpace/DSpace/assets/54576195/3f3d1d32-1c39-42ea-9a59-004839223c92)

The Saxon DocumentPool retained a size over 4GB.

**Comment by ChrisBethgster** [comment]
The DocumentPool of saxon isn't cleared. See [Stackoverflow](https://stackoverflow.com/questions/19764275/java-lang-outofmemoryerror-while-transforming-xml-in-a-huge-directory)

**Comment by ChrisBethgster** [comment]
After a long analysis... And a test with different XSLTs and metadata formats.

The problem doesn't with the original/embedded XSLTs in DSpace. 
In our DSpace we use an extended METS-Format for retro digitized materials with web viewers.
Sample: https://opendata.uni-halle.de/handle/1981185920/84574 || https://opendata.uni-halle.de/oai/dd?verb=GetRecord&metadataPrefix=mets&identifier=oai:opendata.uni-halle.de:1981185920/84574

The XSLT in our METS use the XSLT document(..) function to load an other XML for the output.
The external XMLs are cached in the document pool and increased the memory.

**Comment by ChrisBethgster** [comment]
Minimal sample XSLT:
```
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:doc="http://www.lyncode.com/xoai" version="1.0">
	<xsl:output omit-xml-declaration="yes" method="xml" indent="yes"/>

	<xsl:template match="/">
		<xsl:variable name="metsUri"
			select="doc:metadata/doc:element[@name = 'local']/doc:element[@name = 'mets']/doc:element[@name = 'uri']/doc:element/doc:field[@name = 'value']"/>
		<xsl:copy-of select="document($metsUri)/*"/>
	</xsl:template>
</xsl:stylesheet>
```

local.mets.uri contains a link to a DSpace bitstream with a XML file.

**Comment by ChrisBethgster** [comment]
Issue is also fixed with the new PR in DSpace/xoai#86 and #9045.
